verifier/src/x509.rs

```diff
@@ -100,35 +105,95 @@ impl UnverifiedCertChain {
             ],
             2048,
         );
-        Certificate::verify_with_profile(&self.0, &trust_anchor.0, None, Some(&profile), None)?;
+        Certificate::verify_with_profile(
+            &self.0,
+            &trust_anchor.0,
+            Some(&mut crl.0),
+            Some(&profile),
+            None,
+        )?;
         Ok(VerifiedCertChain(self.0))
     }
-}
 
-/// Try to get a certificate chain from a PEM-encoded string.
-///
-/// # Errors
-/// `Error::MbedTls` if the string is not valid PEM certificate(s).
-impl TryFrom<&str> for UnverifiedCertChain {
-    type Error = Error;
+    /// Try to get a certificate chain from an iterator of PEM encoded strings
+    ///
+    /// # Errors
+    /// `Error::MbedTls` if one of the strings is not valid PEM certificate(s).
+    pub fn try_from_pem<'a, E, I>(pems: I) -> Result<Self>
+    where
+        I: IntoIterator<Item = &'a E>,
+        E: ToString + 'a + ?Sized,
```

Comment:
This signature may feel a bit heavy, but I'm a fan of the way it allows users of the API to do:
```rust
UnverifiedCertChain::try_from_pem(["stuff", "more stuff", "yup"]).unwrap();
```
or
```rust
let string1 = String::from("something");
let string2 = String::from("another");
let string2 = String::from("something else");
let pems = vec![string1, string2, string3];
UnverifiedCertChain::try_from_pem(pems).unwrap();
```

Reply:
I like it.